channels/chan_dahdi.c

```diff
@@ -9191,6 +9191,26 @@ static int dahdi_write(struct ast_channel *ast, struct ast_frame *frame)
 		return -1;
 	}
 
+	if (p->sig == SIG_FXOLS || p->sig == SIG_FXOKS || p->sig == SIG_FXOGS) {
+		struct analog_pvt *analog_p = p->sig_pvt;
+		if (analog_p->callwaitdeluxepending) {
+			unsigned int mssinceflash = ast_tvdiff_ms(ast_tvnow(), analog_p->flashtime);
```

Comment:
I want to question anything that's driven by timing. Does having a 1 second timer suggest that if I'm really quick to do something then I can get a different behavior? What happens if Asterisk (or the system it's hosted on) is faster/slower then expected, could that result in a change in behavior? What if I just can't flash+DTMF within this short window, do I miss out on being able to do something important? One should strive for very predictable results.